src/java.ts

```diff
@@ -51,11 +86,14 @@ export async function getJavaConfig(
   languageClient: SonarLintExtendedLanguageClient,
   fileUri: string
 ): Promise<GetJavaConfigResponse> {
-  const extension: VSCode.Extension<any> | undefined = VSCode.extensions.getExtension('redhat.java');
+  const extension: VSCode.Extension<any> | undefined = getJavaExtension();
   try {
     const extensionApi: any = await extension?.activate();
     if (extensionApi && isJavaApiRecentEnough(extensionApi.apiVersion)) {
       installClasspathListener(languageClient);
+      if (extensionApi.serverMode === ServerMode.LIGHTWEIGHT) {
```

Comment:
We could have a more defensive approach by checking if mode is either 'standard' or 'hybrid', and log that we don't support any other mode. I'm afraid the Java LS adding/renaming modes in the future.